### PR TITLE
Query analysis

### PR DESCRIPTION
**Pull request by stevenj**
No need to actually merge this at this stage.  This PR is mostly for discussion at this stage.  

In it I am documenting the queries and what they are doing.
The only code changes I am making are to indent the query strings themselves for improved clarity.  
(I haven't tested the code changes at all).

### CODE REVIEW

**Comment by minikin** [comment]
Thank you, @stevenj.
I'd propose to merge the documentation part.

**Review by minikin — APPROVED** [review]
LGTM

**Comment by stevenj** [comment]
That's about as good as I can suggest without putting excessive effort into this.  I think we should use these queries as they are better than the original, and solve some problems the originals have because they were not targeting CIP-36 requirements.

See `stake_value_processing.md` for my understanding of how CIP-36 stake value should be calculated.  This may need to change subtly based on what consumes the output and any other requirements for delegation I may not be aware of.

File: src/db/design/vote_registration_query.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        END
+    ) AS purpose
+    ```
+
+    This SQL creates the purpose column. IF the `5` field is present in the
+    metadata, AND its a properly formed integer, it becomes the purpose (as an
+    integer). If it is NOT present, it is defaulted to purpose 0 (Cip-15->36
+    upgrade). Otherwise the purpose is set to -1 to indicate the registration
+    record is invalid and should be ignored.
+
```
Comment (stevenj) [inline src/db/design/vote_registration_query.md:214]: Document that `~` is a regex match operation and `E'` is an "Escape String Constant" ref: https://www.postgresql.org/docs/current/sql-syntax-lexical.html

**Pull request merged**